packages/jni-swig-stub/realm.i

```diff
@@ -88,6 +88,22 @@ std::string rlm_stdstr(realm_string_t val)
     };
 }
 
+// This sets up a type map for all methods with a (partial) argument pattern of:
+//    realm_synchronous_callback_func_t, void* userdata
+// This will make Swig wrap methods taking this argument pattern into:
+//  - a Java method that takes one argument of type `Object` (`jstype`) and passes this object on as `Object` to the native method (`jtype`+``javain`)
+//  - a JNI method that takes a `jobject` (`jni`) that translates the incoming single argument into the actual three arguments of the C-API method (`in`)
```

Comment:
Don't know if we need to replicate the comments for each of these callback methods ... and as mentioned since this is only used for a single method, we could also just do the wrapping manually.

Reply:
The same pattern will be used for `initialData` and migration callbacks, so I think they are worth setting up like this?

Reply:
I guess we CAN actually reuse the type map, as we could just `apply` them on the other signature types and only overwrite the `in` typemap. Just as we have done for the above callbacks that has the `realm_free_userdata_func_t` argument. Then we  can also easily align them all if the C-API ends up adding `realm_free_userdata_func_t` for these calls too.